shared/constants/search.js

```diff
@@ -28,12 +28,12 @@ export type ExtraInfo = {
   fullName: string,
 }
 
-export type SearchResult = {
+export type SearchResult = {|
```

Comment:
Exact object types were necessary to make some dynamic type checking work. We may need to use this pattern more often for disjoint object types.